src/Rings.jl

```diff
@@ -81,6 +81,17 @@ end
 
 *(x::RingElement, y::RingElem) = parent(y)(x)*y
 
+
+"""
+    divexact(x, y)
+
+Return the exact division of `x` by `y`, i.e. an element
+`z` such that `x == yz`; when `x` and `y` do not belong to the same ring,
+they are first coerced into a common ring.
```

Comment:
The following is not a request to change this PR but rather questions about AA it raised in me:

Namely,  guess I am confused as to what the relation between "coercion" and "promotion" in AA is supposed
to be? Are they intended as synonyms? Or perhaps AA wants to avoid "promotion" to make it clear that this differs from promotions in Julia (but then shouldn't this be stated explicitly somewhere? Perhaps it is and I just overlooked it?)

As far as I can tell, the Julia manual only seems to talk about "promotion", no mention of "coerce" or "coercion" is in there. Searching the AA manual for "coercion" yields three sections (for modules and for permutation groups, but e.g. *not* for rings); searching for "promotion" yields a bunch of hits, including https://nemocas.github.io/AbstractAlgebra.jl/latest/rings/#Promotion-rules and the almost identical https://nemocas.github.io/AbstractAlgebra.jl/latest/ncrings/#Promotion-rules (why this duplication???) which talks about "promotion rules" for use in "coercion" but without ever really explaining what "coerce" means.

Reply:
I was knowingly vague here and hoped to be corrected if necessary. My rough idea about promotion vs coercion is that promotion concerns elements of different types, whereas coercion concerns elements with different parents (possibly of the same type). For `divexact`, there might not even be an actual "coercion", but the result is "as if" it happened.

Reply:
Let's turn this into an issue